pkg/controller/common/annotation/controller_version.go

```diff
@@ -138,21 +139,25 @@ func ReconcileCompatibility(ctx context.Context, client k8s.Client, obj runtime.
 // The labels provided must exactly match.
 func checkExistingResources(client k8s.Client, obj runtime.Object, labels map[string]string) (bool, error) {
 
-	accessor := meta.NewAccessor()
-	namespace, err := accessor.Namespace(obj)
+	metaObject, err := meta.Accessor(obj)
 	if err != nil {
-		log.Error(err, "error getting namespace", "kind", obj.GetObjectKind().GroupVersionKind().Kind)
 		return false, err
 	}
+
 	labelSelector := ctrlclient.MatchingLabels(labels)
-	nsSelector := ctrlclient.InNamespace(namespace)
-	// if there's no controller version annotation on the object, then we need to see maybe the object has been reconciled by an older, incompatible controller version
+	nsSelector := ctrlclient.InNamespace(metaObject.GetNamespace())
+
 	var svcs corev1.ServiceList
 	err = client.List(&svcs, labelSelector, nsSelector)
 	if err != nil {
 		return false, err
 	}
-	// if we listed any services successfully, then we know this cluster was reconciled by an old version since any objects reconciled by a 0.9.0+ operator would have a label
```

Comment:
I think this comment was pretty useful

Reply:
It could definitely be better written though, I think I wrote it and it is still confusing to me. Maybe

>// if we listed any services successfully, then we know this owner resource was reconciled by an old version since any owner resources reconciled by a 0.9.0+ operator would have a label already

But it is kind of leaking abstractions and belongs in the caller func IMO. This func just checks for existing resources, so it doesn't necessarily need to only be used for old objects (but it's also unexported so ehhhhhh).